runner/runner.go

```diff
@@ -605,6 +606,54 @@ retry:
 	}
 
 	resp, err := hp.Do(req)
+
```

Comment:
`-status` should work along with other options, unfortunately in this position we are unable to perform retries from `https` to `http`. I would refactor it as follows:

```
resp, err := hp.Do(req)

	fullURL := req.URL.String()

	builder := &strings.Builder{}
	builder.WriteString(stringz.RemoveURLDefaultPort(fullURL))

	if r.options.Status {
		builder.WriteString(" [")
		outputStatus := "success"

		if err != nil {
			outputStatus = "failure"
		}

		switch {
		case !scanopts.OutputWithNoColor && err != nil:
			builder.WriteString(aurora.Red(outputStatus).String())
		case !scanopts.OutputWithNoColor && err == nil:
			builder.WriteString(aurora.Green(outputStatus).String())
		default:
			builder.WriteString(outputStatus)
		}

		builder.WriteRune(']')
	}

	errString := ""
	if err != nil {
		errString = err.Error()
		splitErr := strings.Split(errString, ":")
		errString = strings.TrimSpace(splitErr[len(splitErr)-1])
	}

	if err != nil {
		if !retried && origProtocol == httpx.HTTPorHTTPS {
			if protocol == httpx.HTTPS {
				protocol = httpx.HTTP
			} else {
				protocol = httpx.HTTPS
			}
			retried = true
			goto retry
		}
		return Result{URL: URL.String(), Input: domain, Timestamp: time.Now(), err: err, Failed: err != nil, Error: errString, str: builder.String()}
	}
```